thinc/util.py

```diff
@@ -234,6 +233,13 @@ def to_categorical(
             )
         nongold_prob = label_smoothing / (n_classes - 1)
 
+    if (1 - label_smoothing) < nongold_prob:
+        raise ValueError(
+            f"For {n_classes} number of classes "
+            "label_smoothing parameter has to be less than "
+            f"{1 - nongold_prob}, but found {label_smoothing}."
+        )
+
```

Comment:
Is this correct? The current error message implies that `label_smoothing` should be less than `1 - nongold_prob`, but `nongold_prob` is actually defined in terms of the `label_smoothing` parameter (L234). If you would run this with `label_smoothing` 0.88, wouldn't it say 
> For 5 number of classes label_smoothing parameter has to be less than 0.78, but found 0.88.

?

and if you run it with `label_smoothing` 0.8
> For 5 number of classes label_smoothing parameter has to be less than 0.8, but found 0.8.

Instead, shouldn't we check & report on the value of 
```
(n_classes - 1)/n_classes
```
so that it would consistently report
> For 5 number of classes label_smoothing parameter has to be less than 0.8, but found 0.88.

and

> For 5 number of classes label_smoothing parameter has to be less than 0.8, but found 0.8.

or am I misunderstanding this?

Reply:
No worries. Let's add that second test case to the test suite so we can check the error message for 0.88 as well?